### PR TITLE
Status monitoring / Motor balance / finer mag_decl / GPS HDOP+Home / FR

### PR DESCRIPTION
**Pull request by gaelj**
Motor/Prop balance GUI
Finer control of meg_decl in GUI
Display GPS HDOP
Display GPS Home
Status monitoring (batt, armed, failsafe, calibrating, serial link lost)
FR localization

### CODE REVIEW

File: tabs/motors.css [hunk 1]
```diff
@@ -288,25 +288,49 @@
...
+    float: left;
+}
+
+.tab-motors .motor_testing .motor_rms dd {
+    margin-top: 5px;
+    margin-right: 4px;
+    width: calc((100% / 9) - 10px);
+    text-align: center;
+    float: left;
+    background-color: #00D800;
```
Comment (skaman82) [inline tabs/motors.css]: Please use the correct green color (not a random geen tone) should be #59aa29

File: tabs/motors.js [hunk 2]
```diff
@@ -285,6 +285,16 @@ TABS.motors.initialize = function (callback) {
                 for (var i = 0; i < 3; i++) {
                     if (Math.abs(accel_with_offset[i]) > Math.abs(accel_max_read[i])) accel_max_read[i] = accel_with_offset[i];
                 }
+                
+                for (i = 0; i < 8; i++) {
+                    if ($('div.motor_switches li input').eq(i).is(':checked')) {
+                   	   $('div.motor_rms dd').eq(i).text(rms.toFixed(4));
+                   	   $('div.motor_rms dd').eq(i).css({'background-color' : "#00D800"});
```
Comment (skaman82) [inline tabs/motors.js]: same false green color > should be #59aa29

File: tabs/motors.js [hunk 3]
```diff
@@ -285,6 +285,16 @@ TABS.motors.initialize = function (callback) {
                 for (var i = 0; i < 3; i++) {
                     if (Math.abs(accel_with_offset[i]) > Math.abs(accel_max_read[i])) accel_max_read[i] = accel_with_offset[i];
                 }
+                
+                for (i = 0; i < 8; i++) {
+                    if ($('div.motor_switches li input').eq(i).is(':checked')) {
+                   	   $('div.motor_rms dd').eq(i).text(rms.toFixed(4));
```
Comment (skaman82) [inline tabs/motors.js]: please don't use tabs

**Comment by skaman82** [comment]
Great work @Gaelj :) Starting to review your changes and will give you some more feedback.
Some initial thoughts on styling: please always use the right green color, don't use spaces instead of tabs. The battery status is a good idea, but maybe it could look a little more nicer - maybe like a battery icon on a phone? And the icons under the battery icon have no description at all - so at first nobody will know what the mean. It would be perhaps be better to add a tooltip on hover here.

**Comment by gaelj** [comment]
Thanks for the review ! Sorry about the green color, however it was not random : I picked it from some existing css value that seemed appropriate at the time. I will correct and implement your other suggestions.
Some help with graphics would indeed be great :
- battery icon
- icons for each mode, especially angle & horizon
- the current arm icon could be improved, I guess
- GPS home icon

I'm looking forward to your feedback on the GPS and motors tabs updates ! (IMHO: handy as they are, still plenty of room for improvement)

**Comment by skaman82** [comment]
I know exactly where the color came from - it was the RMS line in the Graph above :) I gladly make some icons for you. Could you make me a screenshot with all visible new icons in the header or send me a hex file for SP F3 board? Can not compile the firmware right now.

**Comment by gaelj** [comment]
https://www.dropbox.com/s/z2g821qp0avsjp5/Capture%20d%27%C3%A9cran%202016-04-12%2000.42.40.png?dl=0

**Comment by gaelj** [comment]
I configured Ecplise to use spaces dammit !! ;-) Something went wrong.
Now I think my edits are full of tabs..... sorry

**Comment by gaelj** [comment]
SP F3 build
https://www.dropbox.com/s/4r63ve4h2f8c9vo/cleanflight_SPRACINGF3.hex?dl=0

**Comment by skaman82** [comment]
Thanks for the screenshot. I thought the will be more icons as I can see right now :) You wrote" Status monitoring (batt, armed, failsafe, calibrating, serial link lost)" 
• Bat is there
• Armed is there 
• Failsafe is there
but what about the other two mentioned things. How they will be shown?

**Comment by skaman82** [comment]
For the tooltip you can use the same way of showing it as it was done her in the receiver tab

bildschirmfoto 2016-04-12 um 00 53 01

**Comment by gaelj** [comment]
- Calibrating is shown by a text change + animation of the corresponding button. We can add this as a top icon as well, for example to show when calibration is triggered externally
- serial link lost: the battery bar appear grey. You can easily see the effect by entering CLI. In the field I connect via Bluetooth so this is a very useful visual cue to me.

**Comment by gaelj** [comment]
Calibration :
https://www.dropbox.com/s/pt1a59lvwi6cfhv/Capture%20d'%C3%A9cran%202016-04-12%2000.58.47.png?dl=0

Lost connection :
https://www.dropbox.com/s/gkf4epx7vz30wh4/Capture%20d%27%C3%A9cran%202016-04-12%2000.59.21.png?dl=0

**Comment by skaman82** [comment]
About calibration: since the calibration status is also show in the log bar only view pixels above, I honestly don't think that showing it in any other way gives us any kind of benefit.

**Comment by skaman82** [comment]
About lost connection: it would be better to have a separate icon for that. I will make one :)

**Comment by gaelj** [comment]
Just to make it clear, we are talking here about the serial connection from CF Configurator to the board.
I would love to show the lost connection status between board and TX but it needs a little more work on CF (I think)

**Comment by skaman82** [comment]
Can we change the color of the battery bar depending on battery capacity left?

**Comment by gaelj** [comment]
It already turns red when voltage goes under the alarm voltage

**Comment by skaman82** [comment]
great! :+1:

**Comment by gaelj** [comment]
I did this after getting a nearly-new 3S lipo down to 7V just fooling around with the PIDs in the garden....

**Comment by gaelj** [comment]
This is a relatively large amount of status icons we are going to add here, as we add one for each mode + lost serial + possibly lost TX (that's 16 = 2 rows of 8 ?). It would also make sense to make them larger than they currently are.
On small screens this will lead to a real-estate problem.
I think it would actually make sense to move the "feature icons" in the Setup tab.
This would require to define a new styling for these buttons, obviously.
Thoughts & suggestions ?

**Comment by skaman82** [comment]
@gaelj  16 icons are way to many. This would make the GUI totally overcrowded. I would suggest only three icons for (armed/failsafe/connection) + bat.

**Comment by gaelj** [comment]
OK, 16 is a lot, I agree. But I still think there is both the space and a need (at least in my case) to show a little extra information.
How about :
- Armed
- Failsafe
- Rate + Angle + Horizon (1 icon position, 3 icons)
- Baro
- Mag
- Headfree + Head adjust (1 position, 2 icons)
- GPS Home + GPS Hold  (1 position, 2 icons)
- Air Mode

That's 2 rows of 4 icons positions.

**Comment by skaman82** [comment]
lay

This is my first shot at this. This barely fits on a 8" Screen. (I would suggest to hide this content for smaller screens). Like I said, it makes no sense to put a huge amount of icons (information) in a main area of the app. Most people do not need this information. If you want to show more status information, I suggest you make a new tab for that an include it in a separate PR. What do you think?

**Comment by skaman82** [comment]
[Icons.zip](https://github.com/cleanflight/cleanflight-configurator/files/225692/Icons.zip)

**Comment by gaelj** [comment]
Excellent job !! Thanks a lot.
Right now I am splitting my modifications into separate branches that will each have it's own PR (now that I have finally understood how to use github properly). Your new graphics are right next on the list.

**Comment by gaelj** [comment]
continued over here: https://github.com/cleanflight/cleanflight-configurator/pull/372

**Pull request closed without merging**